RSocket.Core/Transports/WebSocketTransport.cs

```
@@ -79,11 +79,22 @@ public class WebSocketManager     //https://github.com/aspnet/AspNetCore/blob/ma
 
 					public WebSocketManager(WebSocketTransport transport, CancellationToken cancel = default) { Transport = transport; Cancel = cancel; }
 
-					public async Task<WebSocket> AcceptWebSocketAsync(string subprotocol)     //https://github.com/aspnet/AspNetCore/blob/master/src/Middleware/WebSockets/src/WebSocketMiddleware.cs
+					public async Task<WebSocket> AcceptWebSocketAsync(string subprotocol, IDictionary<string, string> headers)     //https://github.com/aspnet/AspNetCore/blob/master/src/Middleware/WebSockets/src/WebSocketMiddleware.cs
 					{
 						//So in the SignalR code, this is where the WebSocketOptions are actually applied. So junky, so overabstracted. This is why the constructors are inverted so short out all of this madness.
 						var socket = new ClientWebSocket();
-						await socket.ConnectAsync(Transport.Url, Cancel);
+						foreach (string key in headers.Keys)
+						{
+							socket.Options.SetRequestHeader(key, headers[key]);
+						}
+						socket.Options.KeepAliveInterval = new TimeSpan(1, 1, 1);
+						try
```

should not be wrapped in try-catch since the exception is swallowed silently